boost/spirit/home/support/char_encoding/standard_wide.hpp

```diff
@@ -83,7 +83,7 @@ namespace boost { namespace spirit { namespace char_encoding
         strict_ischar(int ch)
         {
             // ch should be representable as a wchar_t
-            return ch >= WCHAR_MIN && ch <= WCHAR_MAX;
+            return ch >= int(WCHAR_MIN) && ch <= int(WCHAR_MAX);
```

Comment:
when `WCHAR_MAX` is `0xffffffffU`, `int(WCHAR_MAX)` is `-1`, so `ch <= int(WCHAR_MAX)` becomes `ch <= -1` which I don't think iswhat we want

Reply:
ah right, thanks for pointing this out. Any idea how to avoid this?

Reply:
remove the test ?
`WCHAR_MIN` and `WCHAR_MAX` seem to span the entire (`unsigned`) `int` range anyway.

Reply:
how about
```
#if WCHAR_MIN==0  //On AARCH64 WCHAR_MIN=0 and WCHAR_MAX=0ffffffffU
  return ch>=0;
#else
  return true;
#endif
```